Civi/Civioffice/EventSubscriber/AbstractCoreEntityCiviOfficeTokenSubscriber.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/*
+ * Copyright (C) 2024 SYSTOPIA GmbH
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation in version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+declare(strict_types = 1);
+
+namespace Civi\Civioffice\EventSubscriber;
+
+use Civi\Core\Event\GenericHookEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+abstract class AbstractCoreEntityCiviOfficeTokenSubscriber implements EventSubscriberInterface {
+
+  /**
+   * @inheritDoc
+   */
+  public static function getSubscribedEvents(): array {
+    return ['civi.civioffice.tokenContext' => 'onCiviOfficeTokenContext'];
+  }
+
+  public function onCiviOfficeTokenContext(GenericHookEvent $event): void {
+    if ($this->getEntityType() === $event->entity_type) {
+      $event->context[$this->getIdContextKey()] = $event->entity_id;
+    }
+    elseif ($this->getEntityType() === ucfirst($event->entity_type)) {
+      trigger_error(
+        sprintf('CiviOffice: Using lower-cased entity types is deprecated. ("%s" was used.)', $event->entity_type),
```

Comment:
Seems we've only had entity types with one word in *CiviOffice*, as they're not only `ucfirst`, but would have to be converted from *snake_case* to *PascalCase* - so I think that's fine for now, but other implementations might have BC problems … Is this e.g. compatible with [this](https://github.com/systopia/funding/blob/59e3e4df4b5fab39b86d463220ce070a34e424ff/Civi/Funding/EventSubscriber/CiviOffice/PaymentInstructionTokenSubscriber.php#L40-L43)?

Reply:
Yes, the linked file isn't affected. It affects only those entity types used in sub classes of `AbstractCoreEntityCiviOfficeTokenSubscriber`.